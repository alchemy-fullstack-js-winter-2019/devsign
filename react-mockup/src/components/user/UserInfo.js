import React from 'react';
import { UserInfoSection } from './UserStyles';
import PropTypes from 'prop-types';

function UserInfo({
  handle,
  profilePicture
}) {
  if(!profilePicture) {
    profilePicture = 'https://res.cloudinary.com/dkrup6iyl/image/fetch/v1552429847/https://res.cloudinary.com/dkrup6iyl/image/upload/v1552429521/download.png';
  }
  
  return (
    <UserInfoSection>
      <img src={profilePicture} alt="user icon" width="100px" />
      <div>
        <h3>{handle}</h3>
        <p>Location, Country</p>
        <p>website: <a href="/" alt="user website">website</a></p>
        <p>Bio. A little bit about user 247.</p>
      </div>
    </UserInfoSection>
  );
}

UserInfo.propTypes = {
  handle: PropTypes.string.isRequired,
  profilePicture: PropTypes.string.isRequired
};

export default UserInfo;
