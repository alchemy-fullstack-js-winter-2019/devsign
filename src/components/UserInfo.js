import React from 'react';
import PropTypes from 'prop-types';
import { getImageUrl } from '../services/image';

// name, profileImage, handle
export default function UserInfo({ user }) {
  const { handle, profileImage, name } = user;
  return (
    <>
    <figure >
      <img alt={`profile image of ${handle}`} src={getImageUrl(profileImage)} />
      {/* <figcaption>{name} {handle}</figcaption> */}
    </figure>
    <aside>
      <ul>
        <li>{name}</li>
        <li>{handle}</li>
      </ul>
    </aside>
    </>

  );
}

UserInfo.propTypes = {
  user: PropTypes.object.isRequired,
};
