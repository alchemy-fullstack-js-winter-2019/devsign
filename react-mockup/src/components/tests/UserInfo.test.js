import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from '../user/UserInfo';
import { MemoryRouter } from 'react-router';

describe('UserInfo snapshot', () => {
  it('can match a snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <UserInfo />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
