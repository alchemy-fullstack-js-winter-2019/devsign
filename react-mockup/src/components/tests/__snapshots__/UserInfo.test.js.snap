// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo snapshot can match a snapshot 1`] = `ShallowWrapper {}`;
